sdk/daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/cctp/MessageSignatureTest.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.data
+package cctp
+
+import java.security.{InvalidKeyException, KeyPairGenerator, SignatureException}
+import java.security.spec.ECGenParameterSpec
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class MessageSignatureTest extends AnyFreeSpec with Matchers {
+
+  "correctly sign and verify secp256k1 signatures" in {
+    val keyPairGen = KeyPairGenerator.getInstance("EC", "BC")
+    keyPairGen.initialize(new ECGenParameterSpec("secp256k1"))
+    val keyPair = keyPairGen.generateKeyPair()
+    val publicKey = keyPair.getPublic
+    val privateKey = keyPair.getPrivate
+    val message = Ref.HexString.assertFromString("deadbeef")
+    val signature = MessageSignatureUtil.sign(message, privateKey)
+
+    MessageSignature.verify(signature, message, publicKey) shouldBe true
+  }
+
+  "invalid public keys fail to verify secp256k1 signatures" in {
+    val keyPairGen = KeyPairGenerator.getInstance("EC", "BC")
+    keyPairGen.initialize(new ECGenParameterSpec("secp256k1"))
+    val invalidKeyPairGen = KeyPairGenerator.getInstance("RSA")
+    invalidKeyPairGen.initialize(1024)
+    val privateKey = keyPairGen.generateKeyPair().getPrivate
+    val invalidPublicKey = invalidKeyPairGen.generateKeyPair().getPublic
+    val message = Ref.HexString.assertFromString("deadbeef")
+    val signature = MessageSignatureUtil.sign(message, privateKey)
+
+    assertThrows[InvalidKeyException](MessageSignature.verify(signature, message, invalidPublicKey))
+  }
+
+  "invalid secp256k1 public keys fail to verify secp256k1 signatures" in {
```

Comment:
nitpick: technically the public key is still valid, just a different one that doesn't match the private key that produced the signature.

```suggestion
  "other secp256k1 public keys fail to verify secp256k1 signatures" in {
```